unified-prototype/unified-plugin/plugin-common/src/main/java/org/gradle/api/experimental/common/ApplicationDependencies.java

```diff
@@ -0,0 +1,18 @@
+package org.gradle.api.experimental.common;
+
+import org.gradle.api.artifacts.dsl.DependencyCollector;
+import org.gradle.api.artifacts.dsl.GradleDependencies;
+import org.gradle.api.plugins.jvm.PlatformDependencyModifiers;
+import org.gradle.api.plugins.jvm.TestFixturesDependencyModifiers;
+import org.gradle.declarative.dsl.model.annotations.Restricted;
+
+/**
+ * The declarative dependencies DSL block for an application.
+ */
+@SuppressWarnings("UnstableApiUsage")
+@Restricted
+public interface ApplicationDependencies extends PlatformDependencyModifiers, TestFixturesDependencyModifiers, GradleDependencies {
+    DependencyCollector getImplementation();
```

Comment:
💭 Does it make sense to still call this "implementation" for an application?  It probably shouldn't revert to "compile" (or should it)?  Is this an opportunity to come up with a new application-only name here and leave impl/api for libraries only?